### ISSUE TITLE
mWeb - VBA - Error message appears after submitting all required info in Onfido flow

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#7270]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go staging.new.expensify.com
2. Log in with expensifail account
3. Go to Settings
4. Tap on any workspace and tap Connect Bank account
5. Use the Plaid flow (Use credentials: User - user_good, Password - pass_good) or manual flow. Both should produce same result
6. Put any information on VBA flow
7. After being taking to Onfido flow, provide any information it asked.
8. When the Onfido flow continues, you're brought back to Expensify with an error

## Expected Result:
After Submitting Onfido information user gets additional info page

## Actual Result:
Error message appears after submitted all required info for Onfido flow

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Mobile Web

**Version Number:** 1.1.30-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
![Bug5409819_Image_from_iOS__3_](https://user-images.githubusercontent.com/43996225/149672010-3f722eb3-b1f4-42af-aaa9-7a95ef1bd194.jpg)
![Bug5409819_Image_from_iOS__3_](https://user-images.githubusercontent.com/43996225/149672018-c1f3ad29-b64a-499f-81ea-aff9bdb58161.png)

https://user-images.githubusercontent.com/43996225/149672021-8c6f731c-c078-4213-a2fb-e29ca2be6587.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @francoisl (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @JmillsExpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by JmillsExpensify** [comment]
Apologies for missing this one! Here's the Upwork job: https://www.upwork.com/jobs/~01442767ed529addbf

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @rushatgabhane (`Exported`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Beamanator (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by JmillsExpensify** [comment]
Still waiting for contributors.

**Comment by eVoloshchak** [comment]
Unable to reproduce. 
Is there something specific about the information you filled out on the previous screens?

https://user-images.githubusercontent.com/9059945/152527368-048a8f37-5362-4444-89ce-ea53674e157e.mp4


Offtop:
I've encountered a strange behaviour when trying to reproduce this issue: if you are on one of the steps for filling out the 'Personal information', pressing the close button on the top right makes the page reload a couple of times.

https://user-images.githubusercontent.com/9059945/152527687-022120e8-405a-421a-ae88-ec52b06860cb.mp4

Is this the intended behaviour or should i report in on Slack?

**Comment by rushatgabhane** [comment]
@eVoloshchak likewise here, unable to reproduce.

> Is this the intended behaviour or should i report in on Slack?

I don't think so. Yes please, report it on slack.

**Comment by rushatgabhane** [comment]
@Beamanator let's close this issue if you can't replicate it either?

**Comment by Beamanator** [comment]
Hmm yeah I'm also unable to reproduce, @kavimuru will you please check this again and close if it's not still an issue? Maybe it got fixed somewhere recently 🤷

**Comment by JmillsExpensify** [comment]
Weird, yeah with two unsuccessful reproductions I'll proactively close now but please feel free to re-open if it crops up again.

**Comment by kbecciv** [comment]
@JmillsExpensify The issue have been reproduced during Regression. 

![IMG_AB45468872AC-1](https://user-images.githubusercontent.com/93399543/153457405-368a8d55-e2a7-40e1-8536-75f8dc7217a4.jpeg)

**Comment by JmillsExpensify** [comment]
Thanks! Let's address this one then. Issue is open for proposals.

**Comment by JmillsExpensify** [comment]
Doubling the rate of this Upwork job to $1,000. Please share your solution here for feedback.

**Comment by JmillsExpensify** [comment]
Going to try putting this issue in #expensify-open-source to see if we can get additional eyes on this one. 👀

**Comment by JmillsExpensify** [comment]
Haven't gotten any interest here so upping the price of this job to $2,000.

**Comment by mallenexpensify** [comment]
D'oh, just updated the title to $2k and [dropped in #expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q/p1647371313143049) but now seeing it's been 8 days since the last double. Let's see if we get any action the next couple days

**Comment by Beamanator** [comment]
Not overdue, waiting for proposals :D

**Comment by JmillsExpensify** [comment]
Still no takers. I am starting to think we should take this internal. Thoughts?

**Comment by mallenexpensify** [comment]
Doubled price to $4k, going to post in #expensify-open-source to get 👀 , also going to post internally to try to get some action there.  https://www.upwork.com/jobs/~0120677daf7eef8539

**Comment by marcaaron** [comment]
Taking this issue Internal for a sec. This is a 3rd party SDK and without a repro a contributor will have a really hard time debugging.

@kbecciv @mvtglobally Do we happen to have a more recent reproduction and the email used when running through the flow? The only things we can do here are:

1. Check the logs to see why Onfido is throwing an error
2. Reach out to Onfido support

My first thought here is that we are testing against the production credentials so also be sure that we are testing against staging.

<img width="389" alt="2022-03-23_14-07-48" src="https://user-images.githubusercontent.com/32969087/159816519-139da5b0-8cdb-4203-ae1f-62f2ef760c14.png">

**Comment by mallenexpensify** [comment]
If needed, Shae is the relationship owner with Onfido, we could ask her to reach out for help.

**Comment by JmillsExpensify** [comment]
We're in the process of considering a new implementation for Onfido (API vs SDK), so I wonder if we should alternatively put this issue on [Hold]. It's still pretty far out admittedly, but I don't think we've gotten a single customer complaint on this, have we?

**Comment by Beamanator** [comment]
@JmillsExpensify I don't think so, and I think it's been pretty hard for us to reproduce. I think that would be fine to put on hold and if there's more complaints and we find an easy way to reproduce we can take it off hold again

**Comment by mallenexpensify** [comment]
Sound like putting on hold is best

**Comment by Beamanator** [comment]
Maybe we can keep on hold 1 more week, if no more complaints let's just close it out

**Comment by JmillsExpensify** [comment]
I'm good with that plan. Will circle back next week and expect to close this :)

**Comment by JmillsExpensify** [comment]
Not exactly the entire week, though no one is going to complain on the weekend anyway. I'm going to close this out as I run through my Friday chores.

**Comment by kavimuru** [comment]
@JmillsExpensify issue was reproducible today when we were executing the PR #10967 So reopening this issue


<img src="https://user-images.githubusercontent.com/43996225/194444640-e4a8b0ea-9560-45f2-88b1-7612aac7f9c6.jpg" width="40%" height="40%">

**Comment by JmillsExpensify** [comment]
Jumping in to test now

**Comment by JmillsExpensify** [comment]
This isn't good. I can't even get to the KYC flow after coming back from Plaid. I just get an infi-spinner. Was able to reproduce twice using steps 1-5 in the OP. 

![IMG_0236](https://user-images.githubusercontent.com/1127863/194641358-ce43a1e9-02b7-4f24-bc8c-c415779b0950.PNG)

**Comment by JmillsExpensify** [comment]
Clicking out of the RHN and navigating back to Connect Bank > Plaid also didn't allow me to get around this.

**Comment by JmillsExpensify** [comment]
I was able to reproduce on iPhone mWeb as well. 
![IMG_5319](https://user-images.githubusercontent.com/1127863/194645845-49aef3a0-7f08-4d45-a75f-adb6a59efc8b.PNG)

**Comment by JmillsExpensify** [comment]
Reaching out to others to see if this is reproducible for them, or something specific to my account.

**Comment by JmillsExpensify** [comment]
Alright, reproducible on other expensifail.com accounts. cc @michaelhaxhiu who was also able to reproduce.

**Comment by michaelhaxhiu** [comment]
Signed into mhaxhiu@expensifail.com > create new workspace > connect bank account > load forever.

<img width=300 src=https://user-images.githubusercontent.com/13206896/194650345-05b5371e-86ca-4547-8729-7fc4e18c2571.png>

**Comment by JmillsExpensify** [comment]
I think this is regrettably a deploy blocker. Throwing the label on.

**Comment by JmillsExpensify** [comment]
Yeah, super weird. Got a couple more testers in the mix, and they were not able to reproduce, so I'm comfortable removing the blocker since this is variable.

**Comment by JmillsExpensify** [comment]
Alright, here's the not good news. I can also reproduce this. 
![IMG_0237 2](https://user-images.githubusercontent.com/1127863/194668050-7f2f9286-40ff-4f08-8f47-28e0093623e1.PNG)

**Comment by JmillsExpensify** [comment]
Also reproducible on production. Not staging specific bug.

**Comment by MariaHCD** [comment]
Catching up here. Looks like there were two issues with the VBBA flow:
1. [Infinite spinner](https://github.com/Expensify/App/issues/7270#issuecomment-1272031594) when after the hand-off back from Plaid: This has had mixed results from different testers. Will look back on this later.

2. Onfido error `Something's gone wrong` after submitting a photo of ID. Looking at the [logs](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%227569d7866cc08e9f-SLC%22)+AND+timestamp:[2022-10-07T20:48:36.735Z+TO+2022-10-07T22:48:36.735Z]&index=logs_expensify-018932):
- We were able to generate the Onfido SDK token successfully:
```
staging-www1.lax | 2022-10-07 21:48:36 735 | jason@expensifail.com | [BankAccountSetup] BankAccount_VerificationAPI - Initiating Onfido flow ~~ bankAccountID: '1729985'

staging-www1.lax | 2022-10-07 21:48:36 735 | jason@expensifail.com | [BankAccountSetup] Onfido_API - Calling Onfido to create applicant

staging-www1.lax | 2022-10-07 21:48:37 057 | jason@expensifail.com | [BankAccountSetup] Onfido_API - Calling Onfido to create web SDK token
```
- So this might be an issue with an Onfido SDK itself 🤔 Will investigate further.

cc: @nkuoch

**Comment by MariaHCD** [comment]
I'm not able to reproduce either issue on staging NewDot 🤔 

1. No infinite spinner after the hand-off from Plaid:

[New Expensify.webm](https://user-images.githubusercontent.com/12268372/194817292-545f9593-2d15-4c7a-a6a4-a4b7baa4b1fc.webm)

2. Able to proceed in the Onfido flow after submitting ID photo:

https://user-images.githubusercontent.com/12268372/194817426-a15971c1-91e6-4cd1-846f-b73e5140da1a.mp4

![Screen Shot 2022-10-10 at 11 28 25 AM](https://user-images.githubusercontent.com/12268372/194817448-0361f6be-a5a5-4ba5-9f32-6e250acdc28b.png)

![Screen Shot 2022-10-10 at 11 28 41 AM](https://user-images.githubusercontent.com/12268372/194817444-7efe2dde-e7b2-4670-8151-f4920feb7ed5.png)

**Comment by MariaHCD** [comment]
The test for https://github.com/Expensify/App/pull/10967 worked as anticipated:


![Screen Shot 2022-10-10 at 11 28 25 AM](https://user-images.githubusercontent.com/12268372/194817448-0361f6be-a5a5-4ba5-9f32-6e250acdc28b.png)

![Screen Shot 2022-10-10 at 11 28 41 AM](https://user-images.githubusercontent.com/12268372/194817444-7efe2dde-e7b2-4670-8151-f4920feb7ed5.png)

![Screen Shot 2022-10-10 at 11 36 10 AM](https://user-images.githubusercontent.com/12268372/194818038-565320a0-96ca-4e1b-a257-ff5b70a02a43.png)

![Screen Shot 2022-10-10 at 11 37 43 AM](https://user-images.githubusercontent.com/12268372/194818033-c4ddfd34-f5f6-4114-bb1b-d190d5e1b284.png)

So perhaps this might be due to inconsistent behaviour from the Onfido SDK on different devices? Perhaps worth reaching out to Onfido support in this case.

**Comment by nkuoch** [comment]
@JmillsExpensify and others who managed to get the errors. Can you still reproduce after you clear all your cookies?

**Comment by MariaHCD** [comment]
Just as a heads up, it doesn't look like the Onfido flow is working as expected on the backend: https://github.com/Expensify/Expensify/issues/232434. This also blocks the App deploy, unfortunately.

Will be investigating that after I've completed the web deploy for today. If we're not able to solve this, I think reverting https://github.com/Expensify/App/pull/10967 to unblock the App deploy would be best.

**Comment by MariaHCD** [comment]
As an update, we have a fix for https://github.com/Expensify/Expensify/issues/232434 but since it's late for me, @nkuoch is taking over. If the [Web PR fix](https://github.com/Expensify/Web-Expensify/pull/35108) is completed and CP'd to prod, the App deploy should be unblocked. If that PR is not completed, we think it's best to revert https://github.com/Expensify/App/pull/10967 and unblock the App deploy, regardless.

**Comment by AndrewGable** [comment]
Revert PR here: https://github.com/Expensify/App/pull/11702

Discussed in Slack here: https://expensify.slack.com/archives/C07J32337/p1665422959225369?thread_ts=1665416806.084149&cid=C07J32337

**Comment by kbecciv** [comment]
We still seen the error message, testing build 1.2.27.0

![IMG_9D0BF3B9BCBD-1](https://user-images.githubusercontent.com/93399543/201438243-cdeb53c0-5694-4e3c-bbc3-ec5d522e72ff.jpeg)

**Comment by roryabraham** [comment]
@maddylewis temperature check on how far out the "use onfido APIs and build out the front-end ourselves" project is? Just wondering how much it's worth investing in bugs related to the Onfido SDK when we're going to be ditching it at some point soon(ish)

**Comment by JmillsExpensify** [comment]
It's a fair question. One thing that worries me about it is that I'm also able to get this in testing pretty easily. This is also a core flow for setting up reimbursement and card, so my gut is that we should fix it again rather than hold out on Onfido's APIs. The HL was just shared, so it's still a decent ways out.

**Comment by Beamanator** [comment]
I'm going to unassign b/c @MariaHCD is also assigned here & she worked on this before I believe - & I'm going OOO for a few days

**Comment by maddylewis** [comment]
@roryabraham -[here](https://github.com/Expensify/Expensify/issues/233363) is the tracking GH for that one. it looks like the high-level DD is out for reviews now. that project is now being led by @bfitzexpensify and @kadiealexander 👍

**Comment by JmillsExpensify** [comment]
I still think we should fix this now. The API solution isn't that close and this looks like a regression essentially since we've only recently been able to reproduce it.

**Comment by nkuoch** [comment]
@kbecciv which email did you use, so I can check the logs?

**Comment by kbecciv** [comment]
@nkuoch Let me check with tester, will update you shortly.

**Comment by kbecciv** [comment]
Tester doesn't remember the credentials.

**Comment by nkuoch** [comment]
There is not much we can do right now without clear reproduction steps, or recent logs of when it happened.
Next time it happens, please make sure to send us the affected email.
Meanwhile, I think we can close it?

**Comment by kbecciv** [comment]
Let me re-check this error in my end

**Comment by kbecciv** [comment]
@nkuoch Reproduced with my own account.
![Screenshot_20221118-093619_Chrome (1)](https://user-images.githubusercontent.com/93399543/202730014-2454fa98-4139-4b3c-a90c-4e709e6f9b24.jpg)

Credentials: applausetester+1107abb1@applause.expensifail.com/Feya86Katya


https://user-images.githubusercontent.com/93399543/202729955-73b3525d-cced-488c-aeae-491b4589e2e9.mp4

**Comment by nkuoch** [comment]
Thanks, will check if I can find anything from the logs

**Comment by nkuoch** [comment]
ApplicantID retrieved from logs: '1c160c9d-0f27-4299-91ef-192ada8920c6'

I've sent an email to Onfido support:
<img width="611" alt="image" src="https://user-images.githubusercontent.com/2463975/203093816-22cc48c1-3cdc-46de-8a68-f0e0a7d71b07.png">

**Comment by JmillsExpensify** [comment]
Thanks so much!

**Comment by JmillsExpensify** [comment]
Any word back from Onfido?

**Comment by nkuoch** [comment]
Nope, just "We are looking into this and would get back to you as soon as possible.". I bumped them.

**Comment by JmillsExpensify** [comment]
Still waiting on Onfido.

**Comment by JmillsExpensify** [comment]
Still waiting on Onfido, it seems.

**Comment by nkuoch** [comment]
They just replied...:

> Hi Nathalie,

> Thank you for getting back to us.
> This error occurs because of the multiple errors that were previously obtained. The document may look like it has gone through validation but in fact it is not. The process continues but additional validation is done when uploading the document.
> When many errors occur in one process flow, the process must continue and sdk validation will determine whether the document can be uploaded or not.
> The errors that appear may not tell you what happened because there are multiple errors happening on the backend.

> I hope this helps.
> We sincerely apologise for the inconvenience caused.
> Please let me know if you have further question or need further assistance and I will be more than glad to help.
```

... So, seems like it happens when taking low quality photos too many times in a row?

@kbecciv can you reproduce it if you pay attention taking a good quality picture right from the start?

**Comment by JmillsExpensify** [comment]
Ugh, shoot. This is why I kept seeing it in my own testing. It's really hard to get a good picture on mobile Safari, which means I ended up needed to try several times before it worked. I'm not certain why, as I have an iPhone 14 Pro. The image quality just sucks compared to the native camera app.

**Comment by JmillsExpensify** [comment]
The tough thing here is that these issues should be solved when we move away from the API, because we can use the native camera and our own file upload/management.

**Comment by JmillsExpensify** [comment]
What's our plan for this issue? See commentary above. After sleeping on it, I could see an argument that this increases the necessity of prioritizing moving to the Onfido API versus the SDK. I don't have any data to back this up, but in my own testing, the chance of having a failed document photo is quite high.

**Comment by JmillsExpensify** [comment]
Also as a heads up, I'm OOO this week. I'll be checking in every couple of days, but my responses will be delayed if you need anything. Please float in #expensify-open-source or #bug-zero if you need anything quickly.

**Comment by nkuoch** [comment]
I'm actually wondering if using their APIs will fix this type of error. Seems like they have a logic where when you send too many times a bad quality photo, the whole flow will fail. (And I guess no matter if it comes from Plaid Link call or Plaid API call?).

As for prioritizing, I'm currently:
- Refactoring and Fixing WAQ bugs for the VBBA setup flow on newDot
- Building tools for chores that are distracting me (like [resetting Plaid tokens](https://github.com/Expensify/Web-Secure/pull/2412) or [refunding fraud settlements](https://github.com/Expensify/Auth/pull/7346/files)) - so engineers don't need to spend time on those anymore.
- I'll be OOO for 2 weeks starting on Wednesday
- After that, Anu was hoping I could be the tech lead on [Cash Back 2.0](https://docs.google.com/document/d/1EB0J9NwpMnnea443pA4dPosFnN1stHSUEctdx_jo7jM/edit?usp=sharing)

**Comment by JmillsExpensify** [comment]
I think this will be fixed because the issue right now is that the images taken via Safari are really bad quality, worse that what you can get out of the native camera app.

**Comment by JmillsExpensify** [comment]
We should align on next steps for this issue in the new year. With WAQ being close to done, we might even put this on hold for the onfido update.

**Comment by nkuoch** [comment]
Yep, let's put it on hold for the onfido update, as there is nothing much we can do meanwhile anyway

**Comment by nkuoch** [comment]
They replied:
<img width="1357" alt="image" src="https://user-images.githubusercontent.com/2463975/211653375-2e1d2028-f6a8-48ea-bace-2fe5f02fffa9.png">

@kbecciv can you tell:
1. What firstname/lastname/email you're using
2. Which device and browser?

Thanks!

**Comment by JmillsExpensify** [comment]
Nice! While we look into this issue, I do think we should pick up the Onfido API project and start moving that forward. WAQ (the project) is over now.

**Comment by kbecciv** [comment]
@nkuoch 
1. Email used: applausetester+0901abb@applause.expensifail.com
2. Device: Samsung Galaxy 21/13

**Comment by nkuoch** [comment]
@kbecciv 
Can you also reproduce it in production, or only on staging? 
Thanks!

**Comment by kbecciv** [comment]
@nkuoch I can not check production, I need to use real bank information.

**Comment by nkuoch** [comment]
you can use random numbers manually (routing number 022000020 and account number 1122334455 for example)

**Comment by nkuoch** [comment]
Nevermind, I managed to reproduce it myself

**Comment by JmillsExpensify** [comment]
@nkuoch should this issue still be open now that you've updated the SDK? I lost track of where we're at, though I think I read somewhere that this was known and only a affected the older SDK, which we've recently upgraded. Is that right, or is there still more work to do here?

**Comment by kadiealexander** [comment]
I think this could be closed. We recently updated to the new SDK and are tracking any issues that arise [here.](https://github.com/Expensify/Expensify/issues/262163). Hopefully the issue in the OP is resolved, but if anyone continues to experience it then that tracking issue is the best place to discuss it. Just so that we can get a gauge for how many issues there are and whether switching to Onfido's API is the best solution.

**Comment by JmillsExpensify** [comment]
Sounds great, thank you!
